base-containers/base/inginious_container_api/utils.py

```diff
@@ -96,3 +96,10 @@ def ssh_wait(ssh_user, timeout=None):
     else:
         print("No one connected !")
         return 253  # timeout
+
+def scripts_isolation(isolate):
+    """ Make the script directory isolated or not from the student """
+    if isolate:
+        os.chmod("student/scripts", 000)
```

Comment:
Did you check this does not fail if the folder does not exist ?

Reply:
Yes, hopefuly I anticipated this one :)
A "student/scripts" directory is always mounted inside the student_container, even if it's empty (and the initial task had no student/scripts directory).
This was just to make it the most general (every student_container thus has a "student/scripts" directory.

Don't hesitate to tell me if you think it's not the right approach

Reply:
LGTM then. I suggest to let @GuillaumeDerval have a look at his.